### PR TITLE
Fix "0" not being printed with print_current_values

### PR DESCRIPTION
**Pull request by thomasaarholt**
### Description of the change
When a value of a component's parameter is 0, it was not being printed by `m.print_current_values()`. Thanks to @erh3cq for bringing it to my attention.

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] add tests,
- [x] ready for review.

### CODE REVIEW

**Comment by thomasaarholt** [comment]
Ready for review. The fix is very minor, but I've also added tests!

**Review by tjof2 — APPROVED** [review]

File: hyperspy/misc/model_tools.py [hunk 1]
```diff
@@ -23,19 +23,29 @@ def _is_iter(val):
...
 class current_component_values():
-    """Convenience class that makes use of __repr__ methods for nice printing in
-     the notebook"""
+    """Convenience class that makes use of __repr__ methods for nice printing in the notebook
+    of the properties of parameters of a component
+     
+    Parameters
+    ----------
+    component : hyperspy component instance
+    only_free : True or False, default False
```
Comment (francisco-dlp) [inline hyperspy/misc/model_tools.py]: ``bool, default False``?

File: hyperspy/misc/model_tools.py [hunk 2]
```diff
@@ -23,19 +23,29 @@ def _is_iter(val):
...
+        If True: Only include the free parameters in the view
+    only_active : True or False, default False
```
Comment (francisco-dlp) [inline hyperspy/misc/model_tools.py]: as above

File: hyperspy/misc/model_tools.py [hunk 3]
```diff
@@ -127,16 +139,24 @@ def _repr_html_(self):
...
-     the notebook"""
-
-    def __init__(self, model, only_free, only_active, component_list=None):
+    """Convenience class that makes use of __repr__ methods for nice printing in the notebook
+    of the properties of parameters in components in a model 
+     
+    Parameters
+    ----------
+    component : hyperspy component instance
+    only_free : True or False, default False
```
Comment (francisco-dlp) [inline hyperspy/misc/model_tools.py]: as above

File: hyperspy/misc/model_tools.py [hunk 4]
```diff
@@ -127,16 +139,24 @@ def _repr_html_(self):
...
+        If True: Only include the free parameters in the view
+    only_active : True or False, default False
```
Comment (francisco-dlp) [inline hyperspy/misc/model_tools.py]: as above

File: hyperspy/misc/model_tools.py [hunk 5]
```diff
@@ -23,19 +23,29 @@ def _is_iter(val):
 
 
 def _iter_join(val):
-    "Joins values of iterable parameters for the fancy view, unless it is None, then blank"
+    "Joins values of iterable parameters for the fancy view, unless it equals None, then blank"
     return "(" + ", ".join(["{:6g}".format(v)
-                            for v in val]) + ")" if val else ""
+                            for v in val]) + ")" if val != None else ""
```
Comment (francisco-dlp) [inline hyperspy/misc/model_tools.py]: stylistically ``is not None`` seems to be recommended over  ``!=None``

File: hyperspy/tests/component/test_component_print_current.py [hunk 6]
```diff
@@ -18,17 +18,84 @@
...
 
 from hyperspy.datasets.example_signals import EDS_SEM_Spectrum
-
+from hyperspy.misc.model_tools import current_component_values, current_model_values
 
 class TestSetParameters:
 
     def setup_method(self):
         self.model = EDS_SEM_Spectrum().create_model()
+        self.component = self.model[1]
```
Comment (francisco-dlp) [inline hyperspy/tests/component/test_component_print_current.py:27]: Where exactly are you testing for the bug fixed in this PR?

**Review by francisco-dlp — CHANGES_REQUESTED** [review]
LGTM except for some very minor style comments.

**Comment by ericpre** [comment]
@thomasaarholt, this one should be quick to finish?

**Comment by thomasaarholt** [comment]
Added @francisco-dlp's comments. Just waiting for tests to finish, should be good to go.

**Comment by thomasaarholt** [comment]
On second look, I realised I hadn't added tests for the three helper functions I'd originally added.

**Pull request merged**